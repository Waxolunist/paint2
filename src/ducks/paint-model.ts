export interface Painting {
  id: number;
  dataUrl: string;
  blobUrl?: string;
}

export interface Point {
  x: number;
  y: number;
}

export interface Stroke {
  points: Point[];
  erase: boolean;
  color: string;
}

export interface PaintingRawData {
  paintingId: number;
  strokes: Stroke[];
}

export interface CRUDPayload {
  painting: Painting;
  rawData: PaintingRawData;
}

export interface PaintState {
  paintings: Painting[];
  activePainting?: CRUDPayload;
}
