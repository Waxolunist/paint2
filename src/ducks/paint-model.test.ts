import {PaintingImpl, PaintingRawData} from './paint-model';

describe('model', () => {
  describe('PaintingImpl', () => {
    it('should set default values', () => {
      const model = new PaintingImpl('data:image/png;base64,abc', 1);
      expect(model.blobUrl).toEqual('');
    });
  });

  describe('PaintingRawData', () => {
    it('should set default values', () => {
      const model = new PaintingRawData(1, []);
      expect(model).toMatchSnapshot();
    });
  });
});
