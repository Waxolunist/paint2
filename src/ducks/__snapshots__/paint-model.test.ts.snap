// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model PaintingRawData should set default values 1`] = `
PaintingRawData {
  "paintingId": 1,
  "strokes": Array [],
}
`;
